### PR TITLE
Create admin page with ability to control featured communities

### LINKED ISSUE #275: [Communities] Give DevHub moderators the ability to set Featured Communities on homepage (FRONT END)
**Problem**
As part of https://github.com/near/neardevhub-contract/issues/36, we made it possible to set the featured communities on the homepage on the contract side. However, admins do not have the ability to set the featured communities through the UI. Currently, they have to rely on developers to modify the smart contract.

**User Story**
As a DevHub admin, I need an easy way to update the features communities on the homepage so that I can showcase various communities without relying on developers to change the code.

**Acceptance Criteria**
- An "Admin" link is visible in the header navigation, visible and accessible only to DevHub admins (not community admins!)
- When a DevHub admin clicks the Admin button, they see a settings page with section for configuring the featured communities
- Within the settings section, DevHub admins can **edit** communities to display on the homepage. The interface for this can be super simple ... just need to solve the goal.
- The UI show a maximum of 5 featured communities, to support the current display limit
- Once the admin clicks save, the changes are instantly reflected on the homepage
- The admin UI should prevent the user from accidentally showing fewer than 4 communities.

Note: It would be great if we can incorporate the label/team permissions functionality (https://github.com/near/neardevhub-widgets/issues/129) into this new admin/settings page. So that there's just one admin page with all the relevant settings.

### PR DESCRIPTION
**Pull request by carina-akaia**
Closes #275

### CODE REVIEW

**Comment by petersalomonsen** [comment]
Is there a preview environment for this? And tests that shows what this PR does?

**Comment by carina-akaia** [comment]
> Is there a preview environment for this? And tests that shows what this PR does?

The link to the demo environment is published within the original ticket

Speaking of tests, I'm not sure if it possible to properly cover the implementation with them, given that page link and content will show up only for accounts with DevHub moderator role. Do you think it makes sense to test the access rejection case?

**Comment by petersalomonsen** [comment]
> > Is there a preview environment for this? And tests that shows what this PR does?
> 
> The link to the demo environment is published within the original ticket
> 
> Speaking of tests, I'm not sure if it possible to properly cover the implementation with them, given that page link and content will show up only for accounts with DevHub moderator role. Do you think it makes sense to test the access rejection case?

In playwright you can manipulate localstorage, and to test what happens for the moderator role, you can simply set the account to `theori.near` like shown in the screenshot below. So you can test both access rejection, and what happens if you have access.

image

**Comment by Tguntenaar** [comment]
@carina-akaia shouldn't we build this in the new root? I was about the start with issue #315, I can move it over but let me know what you prefer.

File: src/gigs-board/components/molecule/tile.jsx [hunk 1]
```diff
@@ -42,7 +49,7 @@ const Tile = ({
   minHeight,
   style,
 }) => (
-  <AttractableDiv
+  <TileRoot
```
Comment (elliotBraem) [inline src/gigs-board/components/molecule/tile.jsx:52]: Is this change related to the story?

File: src/gigs-board/pages/admin.jsx [hunk 2]
```diff
@@ -0,0 +1,264 @@
+/* INCLUDE: "common.jsx" */
```
Comment (elliotBraem) [inline src/gigs-board/pages/admin.jsx:1]: Can we isolate this PR to only be adding the new admin page? And can we have this admin page be rebased to use the new refactor/root, please?

**Review by elliotBraem — CHANGES_REQUESTED** [review]

File: src/gigs-board/pages/admin.jsx (hunk 2 above)
Comment (carina-akaia) [reply src/gigs-board/pages/admin.jsx:1]: Okay, I'll create a separate PR for UIkit adjustments soon, I needed to make some of the components more flexible

File: src/devhub/page/admin.jsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    hints: {
+      disabled: `You can only add ${AdministrationSettings.communities.maxFeatured} communities at a time`,
+    },
+
+    inputProps: { min: 3, max: 40, required: true },
+  },
+};
+
+const featuredCommunities = useQuery("featured_communities");
```
[the remark this answers is not in the document]
Comment (carina-akaia) [reply src/devhub/page/admin.jsx:33]: AC says that the changes sould be applied instantaneously. Plus, state actualization is requirted to preselve consistency, otherwise you may end up rewriting changes made by someone else before you
